v2/pkg/additional/local_stored_collector.go

```diff
@@ -76,7 +75,12 @@ func (o *LocalStorageCollector) AdditionalImagesCollector(ctx context.Context) (
 					domainAndPathComps = transportAndRef[1]
 				}
 				src = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, img.Name}, "/")
-				dst = strings.Join([]string{o.Opts.Destination, img.Name}, "/")
+
+				if isDigestName(img.Name) {
+					dest = strings.Join([]string{o.Opts.Destination, imageShortName(img.Name) + ":" + imageHash(img.Name)[:hashTruncLen]}, "/")
+				} else {
+					dest = strings.Join([]string{o.Opts.Destination, img.Name}, "/")
+				}
 
 				// the following is for having the destination without the initial domain name => later
```

Comment:
@aguidirh - I know this was not part of the PR but could you remove all unwanted comments

Reply:
These will come in handy when we remove the registry from the destination image url
today, we are copying registry.redhat.io/ubi8/ubi to localhost:5000/registry.redhat.io/ubi8/ubi